aqt/addons.py

```diff
@@ -35,7 +35,21 @@ def allAddons(self):
             if not os.path.exists(os.path.join(path, "__init__.py")):
                 continue
             l.append(d)
-        return l
+
+        # os.listdir returns directories in arbitrary order
+        # in case a certain load order need to be enforced,
+        # the numeric "loadorder" property can be added to the
+        # meta json file of an add on.
+
+        loadorder = [
```

Comment:
- it might be preferable to determine `loadorder` during the folder traversal above. This would avoid an additional loop.
- using `dict.get(key, default)` would be slightly more pythonic
- in order to avoid the `zip()` call later down the line you could also use a list of tuples

E.g.:

```python
l = []
for d in os.listdir(self.addonsFolder()):
    path = self.addonsFolder(d)
    if not os.path.exists(os.path.join(path, "__init__.py")):
        continue
    loadorder = self.addonMeta(d).get("loadorder", 9999)
    l.append((loadorder, d))
return [d for _, d in sorted(l)]
```

N.B.: I haven't tested this code in Anki.